validator/keymanager/v2/derived/derived.go

```diff
@@ -78,12 +96,57 @@ func DefaultConfig() *Config {
 
 // NewKeymanager instantiates a new derived keymanager from configuration options.
 func NewKeymanager(
-	ctx context.Context,
+	cliCtx *cli.Context,
 	wallet iface.Wallet,
 	cfg *Config,
 	skipMnemonicConfirm bool,
-	password string,
 ) (*Keymanager, error) {
+	walletFiles, err := wallet.ListDirs()
+	if err != nil {
+		return nil, err
+	}
+	var accountsPassword string
+	if len(walletFiles) == 0 {
+		accountsPassword, err = inputPassword(
+			cliCtx,
+			flags.WalletPasswordFileFlag,
+			newWalletPasswordPromptText,
+			confirmPass,
+			promptutil.ValidatePasswordInput,
+		)
+	} else {
+		validateExistingPass := func(input string) error {
+			if input == "" {
+				return errors.New("password input cannot be empty")
+			}
+			return nil
+		}
+		accountsPassword, err = inputPassword(
+			cliCtx,
+			flags.WalletPasswordFileFlag,
+			walletPasswordPromptText,
+			noConfirmPass,
+			validateExistingPass,
+		)
+	}
+
+	// Check if the wallet seed file exists. If it does not, we initialize one
+	// by creating a new mnemonic and writing the encrypted file to disk.
+	ctx := context.Background()
+	if !fileutil.FileExists(filepath.Join(wallet.AccountsDir(), EncryptedSeedFileName)) {
+		seedConfig, err := initializeWalletSeedFile(accountsPassword, skipMnemonicConfirm)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not initialize new wallet seed file")
+		}
+		seedConfigFile, err := marshalEncryptedSeedFile(seedConfig)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not marshal encrypted wallet seed file")
+		}
+		if err := wallet.WriteEncryptedSeedToDisk(ctx, seedConfigFile); err != nil {
+			return nil, errors.Wrap(err, "could not write encrypted wallet seed config to disk")
+		}
+	}
+
 	seedConfigFile, err := wallet.ReadEncryptedSeedFromDisk(ctx)
```

Comment:
Any way we can prevent the "create a file and write to disk" -> "read file we just made from disk" pattern here?